projects/concurrency/computing-cache/computing-cache.go

```diff
@@ -0,0 +1,37 @@
+package computingCache
+
+import (
+	"concurrency/cache"
+)
+
+type computingCache[K comparable, V any] struct {
+	cache.Cache[K, V]
+	Creator func(K) V
+}
+
+func NewComputingCache[K comparable, V any](entryLimit int, creator func(K) V) computingCache[K, V] {
+	return computingCache[K, V]{
+		Cache:   cache.NewCache[K, V](entryLimit),
+		Creator: creator,
+	}
+}
+
+func Creator[K comparable](key int) any {
+	value := "NewValue"
+	return value
+}
+
+func (c *computingCache[K, V]) Get(key K) V {
+	if currentNode, isExisted := c.M[key]; isExisted {
+		return currentNode.Data.Value
+	}
+	if len(c.M) >= c.Size {
+		deleted, entriesRead := c.L.DeleteFirstNode()
+		delete(c.M, deleted)
+		c.S.TotalReadExisted -= entriesRead
+	}
+	value := c.Creator(key)
+	newNode := c.L.AddNodeToLast(key, value)
+	c.M[key] = newNode
+	return value
```

Comment:
This feels like it's really digging into the internals of your cache (and you had to make the field `L` public to be able to do so). Things like the `L` field, the idea that we have a `map` and a `List` internally, and the relationship between them are very much internal implementation details of our cache type, not part of its public API.

It's interesting to use your existing cache type as a building block here - if you're going to do so, though, you should only use the _public_ API of that type - so you'd instead do something like:

```go
value := c.Creator(key);
c.cache.Put(key, value)
```

In particular, you have a problem here that there's no locking that happens between you deleting an old entry, and you creating a new one. There are ways you can add that locking, but let's skip over thinking about that for now :)

The big challenge with a computing cache is that if two `Get` calls come in at the same time, you only want to do the computing once. Let's imagine we had a `Creator` function which sleeps for 5 seconds and then returns a value, and we have two goroutines trying to get the value for the first time in parallel. We only want to call `Creator` once, not twice.